### PR TITLE
Extract Policy Protection Required error from refresh token errors

### PR DESCRIPTION
**Pull request by midella**

### CODE REVIEW

File: IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m [hunk 1]
```diff
@@ -105,16 +106,36 @@ - (BOOL)verifyResponse:(MSIDAADV1TokenResponse *)response
...
-                                     nil,
-                                     context.correlationId,
-                                     nil);
+            NSMutableDictionary *userInfo = [[NSMutableDictionary alloc] initWithCapacity:1];
+            if (fromRefreshToken.homeAccountId)
+            {
+                [userInfo setObject:fromRefreshToken.homeAccountId forKey:MSIDUserIdKey];
+            }
+
+            if ([MSID_PROTECTION_POLICY_REQUIRED isEqualToString:response.additionalServerInfo[@"suberror"]])
```
Comment (midella) [inline IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m]: The service is currently sending this as "suberror", although the MSID constant is defined as "sub_error"

File: IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m [hunk 2]
```diff
@@ -84,13 +85,13 @@ - (BOOL)verifyResponse:(MSIDAADV1TokenResponse *)response
...
     return [self verifyResponse:response
-               fromRefreshToken:NO
+               fromRefreshToken:nil
                         context:context
                           error:error];
 }
 
 - (BOOL)verifyResponse:(MSIDAADV1TokenResponse *)response
-      fromRefreshToken:(BOOL)fromRefreshToken
+      fromRefreshToken:(MSIDBaseToken<MSIDRefreshableToken> *)fromRefreshToken
```
Comment (jasoncoolmax) [inline IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m]: `(MSIDBaseToken<MSIDRefreshableToken> *)refreshToken` could be better? :)

File: IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m [hunk 3]
```diff
@@ -105,16 +106,36 @@ - (BOOL)verifyResponse:(MSIDAADV1TokenResponse *)response
...
+
+                *error = MSIDCreateError(MSIDOAuthErrorDomain,
+                                         errorCode,
+                                         response.errorDescription,
+                                         response.error,
+                                         nil,
+                                         nil,
+                                         context.correlationId,
+                                         userInfo);
+            }
```
Comment (jasoncoolmax) [inline IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m]: Does it look better if we combine line 117-124 and 130-137 to one call? :)

File: IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m [hunk 4]
```diff
@@ -105,16 +106,34 @@ - (BOOL)verifyResponse:(MSIDAADV1TokenResponse *)response
     {
         if (response.error)
         {
-            MSIDErrorCode errorCode = fromRefreshToken ? MSIDErrorServerRefreshTokenRejected : MSIDErrorServerOauth;
+            NSMutableDictionary *userInfo = [[NSMutableDictionary alloc] initWithCapacity:1];
+            if (refreshToken.homeAccountId)
```
Comment (jasoncoolmax) [inline IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m]: i think we should use `primaryUserId` rather than `homeAccountId` here :)

**Comment by midella** [comment]
The build is broken after merging from dev, but #177 removes the offending lines anyway, so I'll leave this for now to avoid additional merge conflicts.

**Pull request merged**